mathext/hyp2f1.go

```diff
@@ -0,0 +1,466 @@
+package mathext
+
+import (
+	"fmt"
+	"math"
+)
+
+// Machine epsilon used in Cephes.
+var mACHEP float64 = (math.Nextafter(1.0, 2.0) - 1.0) / 2
+
+const eps float64 = 1e-13
+const ethresh float64 = 1e-12
+
+// Hyp2f1 returns the value of the Gaussian Hypergeometric function at z.
+// For |z| < 1, this implementation follows the Cephes library.
+// For |z| > 1, this implementation perform analytic continuation via relevant Hypergeometric identities.
+// See https://en.wikipedia.org/wiki/Hypergeometric_function for more details.
+func Hyp2f1(a float64, b float64, c float64, z float64) (float64, error) {
```

Comment:
Suggest renaming this.
```suggestion
func Hypergeo(a, b, c, z float64) (float64, error) {
```

Can we express the error conditions in-band? The only tenuous cases seem to me to be overflow (expressible as Inf rather than `math.MaxFloat64` and the partial loss of precision (how important is that?)).

Reply:
Good suggestions! `Hypergeo` reads much clearer. Since there're only two tenuous cases, `Inf` and `NaN` represent them nicely.